github/repos_contents_test.go

```diff
@@ -56,17 +56,27 @@ func TestRepositoryContent_GetContent(t *testing.T) {
 		r := RepositoryContent{Encoding: tt.encoding, Content: tt.content}
 		got, err := r.GetContent()
 		if err != nil && !tt.wantErr {
-			t.Errorf("RepositoryContent(%q, %q) returned unexpected error: %v", tt.encoding, tt.content, err)
+			t.Errorf("RepositoryContent(%s, %s) returned unexpected error: %v",
+				stringOrNil(tt.encoding), stringOrNil(tt.content), err)
 		}
 		if err == nil && tt.wantErr {
-			t.Errorf("RepositoryContent(%q, %q) did not return unexpected error", tt.encoding, tt.content)
+			t.Errorf("RepositoryContent(%s, %s) did not return unexpected error",
+				stringOrNil(tt.encoding), stringOrNil(tt.content))
 		}
 		if want := tt.want; got != want {
 			t.Errorf("RepositoryContent.GetContent returned %+v, want %+v", got, want)
 		}
 	}
 }
 
+// stringOrNil converts a potentially null string pointer to string
+func stringOrNil(s *string) string {
+	if s == nil {
+		return "<nil>"
+	}
+	return *s
```

Comment:
random musing - what about making this line `return fmt.Sprintf("%q", *s)` to preserve the quoting?

Reply:
if we want the function call to have quotes around the parameters as in a "real" one, I think we should add them to the `t.Errorf`'s format, and change the surrounding quotes markers: 
```
		if err == nil && tt.wantErr {
			t.Errorf(`RepositoryContent("%s", "%s") did not return unexpected error`,
				stringOrNil(tt.encoding), stringOrNil(tt.content))
		}
```
- this would allow to keep `stringOrNil` doing straightforward conversion to string or "<nil>" without adding additional formatting (surrounding quotes).

Reply:
No, this is not correct because `<nil>` would then be quoted like `"<nil>"` which is not right.
It should either be `<nil>` or `"string-value"`.
But no big deal... we can ignore it.

Reply:
you are right; changed `stringOrNil` to use `fmt.Sprintf("%q", *s)` for quoting